[Term Entry] Kotlin `.ln()` method (#3605)
Codecademy/docs: content/kotlin/concepts/math-methods/terms/ln/ln.md

---
Title: 'ln()'
Description: 'Calculates the natural logarithm of a number.'
Subjects:
  - 'Code Foundation'
  - 'Computer Science'
Tags:
  - 'Math'
  - 'Functions'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The **`.log()`** method in Kotlin's `math` class is used to calculate the natural logarithm of a number.

Reviewer: ```suggestion
The **`ln()`** method in Kotlin's `math` class is used to calculate the natural logarithm of a number.

```

## Syntax

```pseudo
ln(n)
```

> - n: The number of type `double` for which the logarithm is to be calculated.

Reviewer: ```suggestion
- `n`: The number of type `double` for which the logarithm is to be calculated.

```

> The `.ln()` method returns a value of type `double`, the logarithm of `n` to _e_. The following are the special cases:

Reviewer: ```suggestion
The `ln()` method returns a value of type `double`, the logarithm of `n` to _e_. The following values represent special cases:

```

- `ln(NaN)` is `NaN`
- `ln(n)` is `NaN` if `n < 0.0`
- `ln(+Inf)` is `+Inf`
- `ln(0.0)` is `-Inf`

## Example

> This example shows how to use the `.ln()` method to calculate the natural logarithm of `6`:

Reviewer: ```suggestion
This example shows how to use the `ln()` method to calculate the natural logarithm of `6`:

```

---
Title: 'ln()'
Description: 'Calculates the natural logarithm of a number.'
Subjects:
  - 'Code Foundation'
  - 'Computer Science'
Tags:
  - 'Math'
  - 'Functions'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

The **`.log()`** method in Kotlin's `math` class is used to calculate the natural logarithm of a number.

## Syntax

```pseudo
ln(n)
```

- n: The number of type `double` for which the logarithm is to be calculated.

> The `.ln()` method returns a value of type `double`, the logarithm of `n` to `e`. The following are the special cases:

Reviewer 2: ```suggestion
The `.ln()` method returns a value of type `double`, the logarithm of `n` to _e_. The following are the special cases:

```

- `ln(NaN)` is `NaN`
- `ln(n)` is `NaN` if `n < 0.0`
- `ln(+Inf)` is `+Inf`
- `ln(0.0)` is `-Inf`

## Example

This example shows how to use the `.ln()` method to calculate the natural logarithm of `6`:

```kotlin
import kotlin.math.*

fun main(){
    println(ln(6.0))
}
```

The output of this code will be:

```shell
1.791759469228055
```

> ## Codebyte Example

Reviewer 2: Codebytes does not support Kotlin so this needs to be changed to a regular example or removed.
